sections/delegate/Form.tsx

```diff
@@ -0,0 +1,303 @@
+				if (isMounted) setGasLimitEstimate(getNormalizedGasLimit(Number(gasEstimate)));
+			} catch (error) {
+				// console.error(error);
+				if (isMounted) setGasLimitEstimate(null);
+			}
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [getApproveTxData]);
+
+	// already delegated
+	useEffect(() => {
+		if (!delegateApprovalsContract) return;
+		let isMounted = true;
+		(async () => {
+			if (!(properDelegateAddress && action)) return setAlreadyDelegated(false);
+			const alreadyDelegated = await delegateApprovalsContract[
+				GET_IS_APPROVED_CONTRACT_METHODS.get(action)!
+			](address, properDelegateAddress);
+			if (isMounted) setAlreadyDelegated(alreadyDelegated);
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [delegateApprovalsContract, properDelegateAddress, address, action]);
+
+	return (
+		<div data-testid="form">
+			<FormContainer>
+				<Header>
+					<IconButton onClick={onGoBack}>
+						<Svg src={NavigationBack} />
+					</IconButton>
+				</Header>
+
+				<InputsContainer>
+					<AmountInput
+						value={delegateAddress}
+						placeholder={t('delegate.form.input-placeholder')}
+						onChange={onEnterAddress}
+						disabled={false}
+						rows={3}
+						autoComplete={'off'}
+						data-testid="form-input"
+					/>
+				</InputsContainer>
+
+				<SettingsContainer>
+					<ActionSelector {...{ action, setAction }} />
+
+					<SettingContainer>
+						<GasSelector gasLimitEstimate={gasLimit} setGasPrice={setGasPrice} />
+					</SettingContainer>
+				</SettingsContainer>
+			</FormContainer>
+
+			<FormButton
+				onClick={onButtonClick}
+				variant="primary"
+				size="lg"
+				data-testid="form-button"
+				disabled={
+					isWalletConnected &&
+					(!properDelegateAddress || !!buttonState || delegateAddressIsSelf || alreadyDelegated)
+				}
+			>
+				{!isWalletConnected ? (
+					t('common.wallet.connect-wallet')
+				) : (
+					<Trans
+						i18nKey={`delegate.form.button-labels.${
+							buttonState ||
+							(!delegateAddress
+								? 'enter-address'
+								: delegateAddressIsSelf
+								? 'cannot-delegate-to-self'
+								: alreadyDelegated
+								? 'already-delegated'
+								: 'delegate')
+						}`}
+						components={[<NoTextTransform />]}
+					/>
+				)}
+			</FormButton>
+
+			{!error ? null : <ErrorMessage>{error}</ErrorMessage>}
+
+			{txModalOpen && (
+				<TxConfirmationModal
+					onDismiss={() => setTxModalOpen(false)}
+					txError={null}
+					attemptRetry={onButtonClick}
+					content={
+						<TxModalItem>
+							<TxModalItemTitle>{t('delegate.form.tx-confirmation-title')}</TxModalItemTitle>
+							<TxModalItemText>{shortenedDelegateAddress}</TxModalItemText>
+						</TxModalItem>
+					}
+				/>
+			)}
+		</div>
+	);
+};
+
+const Header = styled(FlexDivRowCentered)`
+	justify-content: space-between;
+	width: 100%;
+	padding: 8px;
+`;
+
+const AmountInput = styled.textarea`
```

Comment:
Might be better to use `input` instead of `textarea` and add `text-overflow:ellipsis` css

![image](https://user-images.githubusercontent.com/4015977/114469601-72ac0d80-9bbb-11eb-98ee-6d0d9907a104.png)

Reply:
Gotcha. I think the design required the input to autogrow to rows if need be https://www.figma.com/file/M58E2cxkOgTPWMHRX7FTna/Staking-User-Flows?node-id=3124%3A131

Reply:
Ah okay